### ISSUE TITLE
Open GZ3Doom to homemade headsets with OpenVR

### ISSUE THREAD

**Issue by dethegeek** [cmbruns/gz3doom#140]
Hi

People build many VR headsets with various hardware. I think OpenVR would be a good option to handle many different hardware as this projects aims to provide an abtraction layer between immersive games and software and VR hardware.

You're probably already aware about this : https://github.com/ValveSoftware/openvr . Is there a plan to add OpenVR compatibility ?

**Comment by cmbruns** [comment]
Yes. Issue #121 involves supporting OpenVR. But it's going to take a while. Note also the many open issues with "[Vive]" in the name, These are all implicitly related to the openvr port.

**Comment by dethegeek** [comment]
This sounds really good !

As I quickly looked the API of OpenVR, I understand this may need some time and work. 

the headtracker I'm building emulates a joystick or a mouse ). I configured it to send accelerations to the computer with the hope the game would use the input in a proportional way (with deadzone disabled). Some tests how it really does not work, probably because here is some post-processing in GZDoom for both mouse and joystick.

If I stick my harware to act as a joystick and GZDoom processes this inputs as 3 anges many people would be able to build their own headtracker with a really simple implementation in GZ2Doom as well as an low cost microcontroller (I'm using a Teensy 3.2, featuring joystick HID out of the box) (have a look on this : it is rather close of my project, as I took inspiration from it http://planetkris.com/2014/12/easier-better-arduino-imu-head-tracker)

As the OpenVR implementation will take a while, I'm considering hacking GZ3Doom for this solution. Would you consider this feature interesting ?
